### PR TITLE
Apply TargetDatabaseName in GenerateCreateScript

### PR DESCRIPTION
**Pull request by AroglDarthu**
Apply `TargetDatabaseName`, which is normally used for publishing the dacpac, to the generated create script as well. See issue #186

### CODE REVIEW

File: src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets [hunk 1]
```diff
@@ -216,11 +216,12 @@
       <PostDeploymentScriptArgument>@(PostDeploy->'--postdeploy %(Identity)', ' ')</PostDeploymentScriptArgument>
       <RefactorLogScriptArgument>@(RefactorLog->'--refactorlog %(Identity)', ' ')</RefactorLogScriptArgument>
       <DebugArgument Condition="'$(MSBuildSdkSqlProjDebug)' == 'True'">--debug</DebugArgument>
-      <TreatTSqlWarningsAsErrorsArgument Condition="'$(TreatTSqlWarningsAsErrors)' == 'True' Or '$(TreatWarningsAsErrors)' == 'True'">--warnaserror</TreatTSqlWarningsAsErrorsArgument>
+      <TreatTSqlWarningsAsErrorsArgument Condition="'$(TreatTSqlWarningsAsErrors)' == 'True' Or ('$(TreatWarningsAsErrors)' == 'True' And '$(TreatTSqlWarningsAsErrors)' == '')">--warnaserror</TreatTSqlWarningsAsErrorsArgument>
```
Comment (jeffrosenberg) [inline src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets:219]: In the interests of a clean commit history that's aligned with individual issues, I think this change should be submitted in a separate PR. That's my personal preference, not sure if @jmezach or @ErikEJ have strong feelings either way. 🙂

Comment (AroglDarthu) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets:219]: Makes sense. I'll separate the commits ;-)

**Review by jeffrosenberg — APPROVED** [review]
Looks good to me @AroglDarthu, thanks for your contribution!

File: src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets (hunk 1 above)
Comment (ErikEJ) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets]: Please revert this and do a separate PR

File: README.md [hunk 2]
```diff
@@ -457,7 +457,12 @@ Instead of using `dotnet publish` to deploy changes to a database, you can also
 </Project>
 ```
 
-With this enabled you'll find a SQL script with the name `<datbase-name>_Create.sql` in the bin folder.
+With this enabled you'll find a SQL script with the name `<database-name>_Create.sql` in the bin folder.
+The database name for the create script gets resolved in the following manner:
+1. `TargetDatabaseName`.
+1. Package name.
```
Comment (ErikEJ) [inline README.md:463]: Should it be 1, 2, 3?

File: src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets (hunk 1 above)
Comment (AroglDarthu) [reply src/MSBuild.Sdk.SqlProj/Sdk/Sdk.targets]: I thought I did 😶

**Review by ErikEJ — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (AroglDarthu) [reply README.md:463]: It displays as 1, 2, 3... Don't know what it would add?

1. 1
1. 2
1. 3

vs

1. 1
2. 2
3. 3

File: README.md [hunk 3]
```diff
@@ -457,7 +457,12 @@ Instead of using `dotnet publish` to deploy changes to a database, you can also
 </Project>
 ```
 
-With this enabled you'll find a SQL script with the name `<datbase-name>_Create.sql` in the bin folder.
+With this enabled you'll find a SQL script with the name `<database-name>_Create.sql` in the bin folder.
+The database name for the create script gets resolved in the following manner:
+1. `TargetDatabaseName`.
```
Comment (jmezach) [inline README.md:462]: Isn't this either `TargetDatabaseName` or the name of the project file? I believe those are the only two options right now looking at this change right?

**Comment by jmezach** [comment]
I had one more comment, but this looks good overall. I've just triggered CI to make sure everything is okay here. @AroglDarthu Thank you for your contribution!

**Review by jmezach — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (AroglDarthu) [reply README.md:462]: There is a third condition in `PackageBuilder.GenerateCreateScript` for when the passed in name is null. I don't think that can ever happen, as the package name is always set via `MetadataArguments`. I propose:
- remove the third option from the readme, and
- remove the condition from GenerateCreateScript.

Comment (jmezach) [reply README.md:462]: I guess from the point of view of the command line tool it could possible be null, even though `Sdk.targets` will always provide it. If someone would want to run the command line tool directly it would be nice if they'd now that `--name` is a required argument. I'm fairly sure that System.CommandLine has something for that. If we add that, I'm fine with removing the fallback.

Comment (jmezach) [reply README.md:462]: @AroglDarthu Just checking if you saw my comment here ;).

Comment (AroglDarthu) [reply README.md:462]: I looked into `System.CommandLine`. The docs don't mention `IsRequired`, but fortunately someone filed an [issue](https://github.com/dotnet/command-line-api/issues/1075) about the docs not mentioning that feature 🙈

**Comment by AroglDarthu** [comment]
Made --name REQUIRED for the build command.

File: src/DacpacTool/PackageBuilder.cs [hunk 4]
```diff
@@ -391,9 +391,12 @@ private List<int> ParseSuppressionList(string suppressionList)
 
         public void GenerateCreateScript(FileInfo dacpacFile, string databaseName)
         {
-            databaseName ??= Path.GetFileNameWithoutExtension(dacpacFile.Name);
-            var scriptFileName = $"{databaseName}_Create.sql";
+            if (databaseName is null)
```
Comment (jmezach) [inline src/DacpacTool/PackageBuilder.cs:394]: We should probably use string.IsNullOrWhiteSpace here?

Comment (AroglDarthu) [reply src/DacpacTool/PackageBuilder.cs:394]: Good catch

**Review by jmezach — APPROVED** [review]

**Comment by jmezach** [comment]
Let's get this merged

**Comment by OzBob** [comment]
sorry, test failed
1.16.2 does not generate a .sql file named as per the ```TargetDatabaseName``` node value in the .csproj file

**Comment by AroglDarthu** [comment]
@OzBob Sorry for the confusion, but the `--name` is not meant for `dotnet build --name`. Instead, it is used for `dotnet <dacpactool> build --name`.

**Comment by jmezach** [comment]
@OzBob The plan was to release this together with a couple of other changes as a 2.0 release, but we've hit a bit of a snag with our support for .NET 6 (see #194).

Most of the changes don't really depend on .NET 6 though, so perhaps we should just release it as is. @ErikEJ @jeffrosenberg Thoughts?

**Comment by ErikEJ** [comment]
.NET 6 support could be a while from now, so maybe just release this?

**Comment by OzBob** [comment]
I'd really appreciate it added to the templates nuget, as I'm intending to use this methodology to generate SQL scripts against a series of customer databases and the latest dacpac. Hopefully as a VSDevOps pipeline and/or Github Actions too.

**Comment by jmezach** [comment]
I've just released this PR together with #188, #189 and #190 as version 1.17.0.

**Comment by OzBob** [comment]
Confirmed: https://www.nuget.org/packages/MSBuild.Sdk.SqlProj.Templates/ is at v1.17.0
